src/main/java/me/arminb/sara/dao/QuestionDAOImpl.java

```diff
@@ -0,0 +1,181 @@
+        question.setUser(questionDoc.get("user").toString());
+
+        List<Answer> ansList = (List<Answer>) questionDoc.get("answers");
+        question.setAnswers(ansList);
+
+        return question;
+    }
+
+    @Override
+    public Question findById(String id) throws DataAccessException {
+        try {
+            MongoCollection<Document> collection = database.getCollection("questions");
+            BasicDBObject query = new BasicDBObject();
+            query.append("_id", new ObjectId(id));
+            MongoCursor<Document> cursor = collection.find(query).iterator();
+            if (cursor.hasNext()) {
+                Document questionDoc = cursor.next();
+                Question question = mapQuestion(questionDoc);
+                return question;
+            }
+            else{
+                return null;
+            }
+        } catch (MongoException e) {
+            logger.warn("Failed to fetch question {} from the database", id, e);
+            throw new DataAccessException();
+        }
+    }
+
+    @Override
+    public List<Question> find(String title, String tag, Integer pageNumber, Integer pageCount) throws DataAccessException {
+        if (pageNumber == null){
+            pageNumber = DEFAULT_PAGE_NUMBER;
+        }
+        if (pageCount == null){
+            pageCount = DEFAULT_PAGE_COUNT;
+        }
+        List<Question> list = new ArrayList();
+        try {
+            MongoCollection<Document> collection = database.getCollection("questions");
+            BasicDBObject query = new BasicDBObject();
+            if (title != null) {
+                query.append("title", title);
+            }
+            if (tag != null) {
+                query.append("tags", tag);
+            }
+            MongoCursor<Document> cursor = collection.find(query).skip(pageCount * (pageNumber - 1)).limit(pageCount).iterator();
+            while (cursor.hasNext()) {
+                Document questionDoc = cursor.next();
+                Question question = mapQuestion(questionDoc);
+                list.add(question);
+            }
+            if (list.size() != 0) {
+                return list;
+            }
+            else {
+                return null;
+            }
+        } catch (MongoException e) {
+            logger.warn("Failed to find questions on the database", e);
+            throw new DataAccessException();
+        }
+    }
+
+    @Override
+    public Question saveQuestion(Question question) throws DataAccessException {
+        try{
+            MongoCollection<Document> collection = database.getCollection("questions");
+
+            if (question.getId() == null){
+                question.setId(new ObjectId().toString());
+                question.setCreatedAt(new Date());
+                Document doc = new Document("_id", new ObjectId(question.getId())).append("title", question.getTitle()).append("rate", question.getRate())
```

Comment:
ceratedAt doesn't need to be saved in the database. Also, use modified_at instead of modified_date